diff --git a/assets/esm_shims.js b/assets/esm_shims.js
index c238cb3317b5300d9fc7b3672ad859241f9bfd58..ad1571b1b3b65eeb5dde7c4622e9f6baf637f117 100644
--- a/assets/esm_shims.js
+++ b/assets/esm_shims.js
@@ -1,9 +1,12 @@
 // Shim globals in esm bundle
-import { fileURLToPath } from 'url'
-import path from 'path'
+import { fileURLToPath } from "url";
+import path from "path";
+import { createRequire } from "module";
 
-const getFilename = () => fileURLToPath(import.meta.url)
-const getDirname = () => path.dirname(getFilename())
+const getFilename = () => fileURLToPath(import.meta.url);
+const getDirname = () => path.dirname(getFilename());
 
-export const __dirname = /* @__PURE__ */ getDirname()
-export const __filename = /* @__PURE__ */ getFilename()
+export const __dirname = /* @__PURE__ */ getDirname();
+export const __filename = /* @__PURE__ */ getFilename();
+
+globalThis.require = createRequire(import.meta.url);
